crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -798,6 +798,10 @@ unsafe impl SystemParamState for WorldState {
 ///
 /// A local may only be accessed by the system itself and is therefore not visible to other systems.
 /// If two or more systems specify the same local type each will have their own unique local.
+/// If multiple [`SystemParam`]s within the same system each specify the same local type
+/// each will get their own local.
+///
+/// The supplied lifetime parameter is the [`SystemParam`]s `'s` lifetime.
```

Comment:
IMO this comment is unnecessary, simply renaming the lifetime to `'s` is good enough.

Reply:
While just having `'s` does express the information it expresses it in a manner that is easy to miss. And given the cost of missing that information

> Users can get complicated incorrect suggested fixes if they pass the wrong lifetime

I think it is worth repeating once in a more explicit visible fashion.

Reply:
Still not convinced that it's necessary, since that comment isn't adding any information that you can't gather just from glancing at the type.

This isn't a significant complaint though, so I won't block on it or anything.